plugin/build.gradle.kts

```diff
@@ -18,13 +18,14 @@ pluginBundle {
 }
 
 dependencies {
-    compileOnly("com.google.cloud.tools:jib-gradle-plugin-extension-api:0.4.0")
-    compileOnly("gradle.plugin.com.google.cloud.tools:jib-gradle-plugin:3.1.1")
+    implementation("com.google.cloud.tools:jib-gradle-plugin-extension-api:0.4.0")
```

Comment:
We should avoid adding these as transitive dependencies and instead keep them in the `compileOnly`  and `testImplementation` scopes. This is tricky because of how the gradle test kit works in the functional test suite, but we can configure it in the gradle script to work.


```kotlin
val plugin: Configuration by configurations.creating

configurations.compileOnly {
    extendsFrom(plugin)
}

configurations.testImplementation {
    extendsFrom(plugin)
}



dependencies {
    plugin("com.google.cloud.tools:jib-gradle-plugin-extension-api:0.4.0")
    plugin("gradle.plugin.com.google.cloud.tools:jib-gradle-plugin:3.1.1")

  / // rest of the deps
}

tasks.named<PluginUnderTestMetadata>("pluginUnderTestMetadata") {
    // adds dependencies with the plugin configuration to the plugin classpath
    pluginClasspath.setFrom(pluginClasspath.plus(plugin.files))
    // avoid warnings
    dependsOn(tasks.compileKotlin)
    dependsOn(tasks.compileJava)
    dependsOn(tasks.processResources)
}


// ...


// Add a task to run the functional tests
val functionalTest by tasks.registering(Test::class) {
    testClassesDirs = functionalTestSourceSet.output.classesDirs
    classpath = functionalTestSourceSet.runtimeClasspath
    // functional test task needs these task deps explicitly
    dependsOn(tasks.compileKotlin)
    dependsOn(tasks.processResources)
}
```

This enables us to still have a pom for the plugin with minimal dependencies and sets the responsibility on the user of the plugin to bring the jib plugin along with them (which is important because not everyone using the plugin jar is also using jib)

Reply:
Thanks for the suggestion, it seems to have done the trick.